src/cc/lib/graph/partition.cc

```diff
@@ -283,9 +283,18 @@ Type Partition::GetNodeType(uint64_t internal_node_id) const
 {
     return m_node_types[internal_node_id];
 }
-void Partition::GetNodeFeature(uint64_t internal_id, std::span<snark::FeatureMeta> features,
+
+bool Partition::HasNodeFeatures(uint64_t internal_node_id) const
+{
+    return GetNodeType(internal_node_id) != snark::DEFAULT_NODE_TYPE;
+}
+
+bool Partition::GetNodeFeature(uint64_t internal_id, std::span<snark::FeatureMeta> features,
                                std::span<uint8_t> output) const
 {
+    if (!HasNodeFeatures(internal_id))
```

Comment:
Why do we have to check feature presence by the node type instead of feature index? On line 307 we fill `output` with 0s, can we return false from the method and fill zeros in the client? Also less encoding/decoding in protobufs..

Reply:
We don't need node_type to distinguish that node doesn't have any features. The problem with current approach is it doesn't allow a node to have features on different partitions. To resolve it we need to merge on the client, instead of filling zeros on the server. However, it is not related to this PR and will be fixed later with temporal feature.